en/ros/external_position_estimation.md

```diff
@@ -1,114 +1,220 @@
+  ```bash
+  roslaunch vrpn_client_ros sample.launch server:=<mocap machine ip>
+  ```
+
+If you named the rigidbody as `robot1`, you will get a topic like `/vrpn_client_node/robot1/pose`
+
+
+### Relaying Pose Data to PX4
+
+MAVROS has plugins to relay a visual estimation from a VIO or MoCap system using the following pipelines:
+
+ROS | MAVLink | uORB 
+--- | --- | ---
+/mavros/vision_pose/pose | [VISION_POSITION_ESTIMATE](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE) | `vehicle_visual_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_VISION_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_VISION_NED)) | `vehicle_visual_odometry`
+/mavros/mocap/pose | [ATT_POS_MOCAP](https://mavlink.io/en/messages/common.html#ATT_POS_MOCAP) | `vehicle_mocap_odometry`
+/mavros/odometry/odom | [ODOMETRY](https://mavlink.io/en/messages/common.html#ODOMETRY) (`frame_id =` [MAV_FRAME_MOCAP_NED](https://mavlink.io/en/messages/common.html#MAV_FRAME_MOCAP_NED)) | `vehicle_mocap_odometry`
+
+You can use any of the above pipelines with LPE. 
 
-> **Tip**If performance is still poor, try increasing the `LPE_PN_V` parameter. This will cause the estimator to trust measurements more during velocity estimation..
+If you're working with EKF2, only the "vision" pipelines are supported. 
+To use MoCap data with EKF2 you will have to [remap](http://wiki.ros.org/roslaunch/XML/remap) the pose topic that you get from MoCap:
+- MoCap ROS topics of type `geometry_msgs/PoseStamped` or `geometry_msgs/PoseWithCovarianceStamped` must be remapped to `/mavros/vision_pose/pose`. 
+  The `geometry_msgs/PoseStamped` topic is most common as MoCap doesn't usually have associated covariances to the data.
+- If you get data through a `nav_msgs/Odometry` ROS message then you will need to remap it to `/mavros/odometry/odom`.
+- TBD from `/vrpn_client_node/<rigid_body_name>/pose` to `/mavros/vision_pose/pose`. 
 
-## Asserting on Reference Frames
 
-This section shows how to setup the system with the proper reference frames. There are various representations but we will use two of them: ENU and NED. 
+## Asserting on Reference Frames 
 
-* ENU has a ground-fixed frame where *x* axis points East, *y* points North and *z* up. Robot frame is *x* towards the front, *z* up and *y* accordingly.
+This section shows how to setup the system with the proper reference frames. 
+There are various representations but we will use two of them: ENU and NED. 
 
-* NED has *x* towards North, *y* East and *z* down. Robot frame is *x* towards the front, *z* down and *y* accordingly.
+* ENU has a ground-fixed frame where *x* axis points East, *y* points North and *z* up.
+  Robot body frame is *x* towards the front, *z* up and *y* accordingly.
+* NED has *x* towards North, *y* East and *z* down.
+  Robot body frame is *x* towards the front, *z* down and *y* accordingly.
```

Comment:
Does this mean "Robot body frame when in an ENU context"? I think it is worth explaining why we have defined Robot body frame twice.

Reply:
Check https://github.com/PX4/Devguide/pull/681/files#r242068527